src/rsa/padding.rs

```diff
@@ -154,3 +155,150 @@ impl Verification for PKCS1 {
         })
     }
 }
+
+/// PSS Padding as described in https://tools.ietf.org/html/rfc3447#section-9.1.
+/// It generates a random salt equal in length to the output of the specified
+/// digest algorithm and uses MGF1 as the mask generating function.
+pub struct PSS {
+    digest_alg: &'static digest::Algorithm,
+}
+
+// The maxmimum supported output length for PSS padding is equal to the maximum
+// supported RSA modulus length.
+// TODO: Can we avoid requiring this MAX_OUTPUT_LEN buffer?
+const MAX_OUTPUT_LEN: usize = 8192 / 8;
+
+// Fixed prefix used in the computation of PSS encoding and verification.
+const PSS_PREFIX_ZEROS: [u8; 8] = [0u8; 8];
+
+impl Verification for PSS {
+    // PSS verification as specified in
+    // https://tools.ietf.org/html/rfc3447#section-9.1.2
+    fn verify(&self, msg: untrusted::Input, encoded: untrusted::Input,
+              bit_len: usize) -> Result<(), error::Unspecified> {
+        // Number of bytes required for bit_len bits.
+        let em_len = 1 + (bit_len - 1) / 8;
+        assert_eq!(encoded.len(), em_len);
+        // Create a bit mask to match the size of the modulus used.
+        let top_byte_mask = (0xffu16 >> 1 + bit_len % 8) as u8;
```

Comment:
We need to have unit tests for this particular calculation, with vectors for each possible value of `bit_len % 8`. These tests don't need to do the RSA part; they just need to call this function and/or call a function that does just this calculation.

Clippy says:
> operator precedence can trip the unwary. Consider parenthesizing your expression: `0xffu16 >> (1 + bit_len % 8)`,

That sounds like a good recommendation.